pipelined/bevy_sprite2/src/sprite.rs

```diff
@@ -4,7 +4,20 @@ use bevy_reflect::{Reflect, TypeUuid};
 #[derive(Debug, Default, Clone, TypeUuid, Reflect)]
 #[uuid = "7233c597-ccfa-411f-bd59-9af349432ada"]
 #[repr(C)]
-pub struct Sprite {
+pub struct Sprite2d {
+    /// Flip the sprite along the X axis
+    pub flip_x: bool,
+    /// Flip the sprite along the Y axis
+    pub flip_y: bool,
+    /// An optional custom size for the sprite that will be used when rendering, instead of the size
+    /// of the sprite's image
+    pub custom_size: Option<Vec2>,
```

Comment:
Now that `Sprite2d`/`Sprite3d` is a component inside the sprite sheet (texture-atlas-based) bundles, this `custom_size` field should be expected to work for those too.

It seems like it is currently broken, because I did not see this being handled in `extract_atlases_{2d,3d}`? You should fix those functions to account for this field, similar to `extract_sprites_{2d,3d}`.

P.S: also, I don't understand how sprite flipping is implemented, so I couldn't check that, but it's worth making sure that `flip_x`/`flip_y` are also handled correctly, for both standalone sprites as well as atlas/spritesheet sprites. Don't forget to test that. :)

Reply:
I'll look into it. Maybe I've overlooked this.

Reply:
IIRC, sprite flipping is not available for the new rendering pipeline atm. But I will also take a look at that.

Reply:
Flipping can be achieved in two ways if im correctly. Either swap the uv coords or flip the quad by 180 degrees (since sprites are not culled) in the given direction. I don't know whats the proper implementation, though.